test/curve-test.js

```diff
@@ -162,3 +162,83 @@ describe('Curve', function() {
     assert(p.eq(neg2));
   });
 });
+
+describe('Point codec', function () {
+  var shortCurve = elliptic.curves.secp256k1.curve;
+  var montCurve = elliptic.curves.curve25519.curve;
+
+  var shortPointEvenY = {
+    coordinates: {
+      x: "79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798",
+      y: "483ada7726a3c4655da4fbfc0e1108a8fd17b448a68554199c47d08ffb10d4b8",
+    },
+    compactEncoded:
+      "02" +
+      "79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798",
+    encoded:
+      "04" +
+      "79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798" +
+      "483ada7726a3c4655da4fbfc0e1108a8fd17b448a68554199c47d08ffb10d4b8"
+  };
+
+  var shortPointOddY = {
+    coordinates: {
+      x: "fff97bd5755eeea420453a14355235d382f6472f8568a18b2f057a1460297556",
+      y: "ae12777aacfbb620f3be96017f45c560de80f0f6518fe4a03c870c36b075f297",
+    },
+    compactEncoded:
+      "03" +
+      "fff97bd5755eeea420453a14355235d382f6472f8568a18b2f057a1460297556",
+    encoded:
+      "04" +
+      "fff97bd5755eeea420453a14355235d382f6472f8568a18b2f057a1460297556" +
+      "ae12777aacfbb620f3be96017f45c560de80f0f6518fe4a03c870c36b075f297"
+  };
+
+  function makeShortTest(definition) {
+    var curve = shortCurve;
+    return function() {
+      var co = definition.coordinates;
+      var p = curve.point(co.x, co.y);
+
+      // Encodes as expected
+      assert.equal(p.encode(false, 'hex'), definition.encoded);
+      assert.equal(p.encode(true, 'hex'), definition.compactEncoded);
+
+      // Decodes as expected
+      assert(curve.decodePoint(definition.encoded, 'hex').eq(p));
+      assert(curve.decodePoint(definition.compactEncoded, 'hex').eq(p));
```

Comment:
Because we are using a string, we route to utils.toArray, which interestingly seems to default to assuming each character in the string decodes to a byte. `hex` must explicitly be passed along. I wonder if it would make more sense to default to `hex` for string inputs ?